src/core/load-fragment.c

```diff
@@ -6573,3 +6575,49 @@ int config_parse_tty_size(
 
         return config_parse_unsigned(unit, filename, line, section, section_line, lvalue, ltype, rvalue, data, userdata);
 }
+
+int config_parse_regex(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+#if HAVE_PCRE2
+        _cleanup_(sym_pcre2_code_freep) pcre2_code *cs = NULL;
+        char **regex = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        r = dlopen_pcre2();
+        if (r < 0)
+                return r;
+
+        if (isempty(rvalue)) {
+                *regex = mfree(*regex);
+                return 0;
+        }
+
+        r = pattern_compile(rvalue, 0, &cs);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid regex pattern, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        free_and_strdup(regex, rvalue);
+#else
+        if (!isempty(rvalue)) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0, "PCRE2 support is not compiled in, ignoring: %s", lvalue);
+                return 0;
+        }
+#endif
+
+        return 0;
+}
```

Comment:
files should end in a newline. this one ends with a regular character. Something is off with your editor